### PR TITLE
Add validation for controller using FluentValidator

### PR DESCRIPTION
**Pull request by MaliyOna**

### CODE REVIEW

**Review by nantonov — APPROVED** [review]

File: CatNote.API/Validators/AchievementDTOValidator.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public AchievementDTOValidator()
+    {
+        RuleFor(x => x.Title)
+            .NotEmpty()
+            .NotNull();
+
+        RuleFor(x => x.Description)
+            .NotEmpty()
+            .NotNull();
```
Comment (SasukeVapes) [inline CatNote.API/Validators/AchievementDTOValidator.cs:16]: Might wanna add .NotEmpty().WithMessage("Title Required")

File: CatNote.API/Validators/TaskDTOValidator.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public TaskDTOValidator()
+    {
+        RuleFor(x => x.Title)
+            .NotEmpty()
+            .NotNull();
+
+        RuleFor(x => x.Status)
+            .NotEmpty()
+            .NotNull();
+
```
Comment (SasukeVapes) [inline CatNote.API/Validators/TaskDTOValidator.cs:17]: Same here

File: CatNote.API/Validators/TaskShortDTOValidator.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace CatNote.API.Validators;
+
+public class TaskShortDTOValidator : AbstractValidator<TaskShortDTO>
+{
+    public TaskShortDTOValidator()
+    {
+        RuleFor(x => x.Title)
+            .NotEmpty()
+            .NotNull();
```
Comment (SasukeVapes) [inline CatNote.API/Validators/TaskShortDTOValidator.cs:12]: Same here

File: CatNote.API/Validators/UserDTOValidator.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace CatNote.API.Validators;
+
+public class UserDTOValidator : AbstractValidator<UserDTO>
+{
+    public UserDTOValidator()
+    {
+        RuleFor(x => x.UserName)
+            .NotEmpty()
+            .NotNull();
```
Comment (SasukeVapes) [inline CatNote.API/Validators/UserDTOValidator.cs:12]: And here

**Review by SasukeVapes — APPROVED** [review]

**Review by Finka95 — APPROVED** [review]

**Pull request merged**